projects/temporal-bargraph/src/lib/temporal-bargraph.component.ts

```diff
@@ -32,19 +32,31 @@ export class TemporalBargraphComponent implements VisualizationComponent,
 
   spec: Spec;
   options: Options = { renderer: 'svg' };
+  expanded = false;
+
+  optionsHidden = true;
 
   private nodes: TDatum<VisualizationNode>[] = [];
   private nodesSubscription: Subscription;
-
+  
   constructor(private dataProcessorService: DataProcessorService) { }
 
+  togglePanel() {
+    this.optionsHidden = !this.optionsHidden;
+  }
+
+  toggleExpanded() {
+    this.expanded = !this.expanded;
+    this.updateSpec()
+  }
+
   updateSpec(): void {
     this.spec = temporalBargraphSpec({
       hasYOrder: !!this.data?.graphicSymbols['bars']?.graphicVariables?.hasOwnProperty('y-order'),
       ...this.propertyDefaults,
```

Comment:
expanded should be part of the propertyDefaults object. See how `enableZoomPan` option is used in Geomap

Reply:
It may not be obvious, but this is how the options are serialized/saved to the project AND how a project can set the options used when the visualization is rendered.